store/postgres/src/entities.rs

```diff
@@ -0,0 +1,993 @@
+            Table::Public(_) => {
+                let query = format!("
+                    create index if not exists {name} on public.entities 
+                    using {index_type} ((data->'{attribute_name}'{jsonb_operator}'data') {index_operator})
+                    where subgraph='{subgraph}' and entity='{entity_name}'",
+                    name=&index.index_name,
+                    index_type=index_type,
+                    attribute_name=&index.attribute_name,
+                    jsonb_operator=jsonb_operator,
+                    index_operator=index_operator,
+                    subgraph=index.subgraph_id.to_string(),
+                    entity_name=&index.entity_name);
+                conn.batch_execute(&*query)?;
+                Ok(1)
+            }
+            Table::Split(entities) => {
+                let name = format!(
+                    "entities_{}_{}",
+                    to_snake_case(&index.entity_name),
+                    to_snake_case(&index.attribute_name)
+                );
+                let query = format!("
+                    create index if not exists {name} on {subgraph}.entities 
+                    using {index_type} ((data->'{attribute_name}'{jsonb_operator}'data') {index_operator})
+                    where entity='{entity_name}'",
+                    name=name,
+                    subgraph=entities.schema,
+                    index_type=index_type,
+                    attribute_name=&index.attribute_name,
+                    jsonb_operator=jsonb_operator,
+                    index_operator=index_operator,
+                    entity_name=&index.entity_name);
+                conn.batch_execute(&*query)?;
+                Ok(1)
+            }
+        }
+    }
+}
+
+/// Delete all entities. This function exists solely for integration tests
+/// and should never be called from any other code. Unfortunately, Rust makes
+/// it very hard to export items just for testing
+pub fn delete_all_entities(conn: &PgConnection) -> Result<usize, StoreError> {
+    // Delete public entities and related data
+    let mut rows = diesel::delete(public::entities::table).execute(conn)?;
+    rows = rows + diesel::delete(public::entity_history::table).execute(conn)?;
+    rows = rows + diesel::delete(public::event_meta_data::table).execute(conn)?;
+    // Delete subgraphs entities
+    rows = rows + diesel::delete(subgraphs::entities::table).execute(conn)?;
+    Ok(rows)
+}
+
+pub(crate) fn create_schema(
+    conn: &PgConnection,
+    subgraph_id: &SubgraphDeploymentId,
+) -> Result<(), StoreError> {
+    // Create a schema for the deployment
+    let schemas: Vec<String> = diesel::insert_into(sgd_schemas::table)
+        .values(sgd_schemas::subgraph.eq(subgraph_id.to_string()))
+        .returning(sgd_schemas::name)
+        .get_results(conn)?;
+    let schema_name = schemas
+        .first()
+        .ok_or_else(|| format_err!("failed to read schema name for {} back", subgraph_id))?;
```

Comment:
Our error messages start with a capital letter.

Reply:
There's more cases like this in this and other files.

Reply:
Though the api guidelines says they should be like this, imo we should switch our convention to what the guidelines say https://rust-lang-nursery.github.io/api-guidelines/interoperability.html#examples-of-error-messages.

Reply:
OK, I am willing to relax my urge to begin all sentences we present to users with a capital letter. ;)